apps/site/assets/ts/schedule/components/schedule-finder/UpcomingDepartures.tsx

```diff
@@ -152,37 +218,77 @@ const UpcomingDeparturesHeader = (
   </div>
 );
 
-export const UpcomingDepartures = ({
-  state,
-  input
-}: Props): ReactElement<HTMLElement> | null => {
-  const { data: journeys, error, isLoading } = state;
+export const fetchData = async (
+  input: UserInput,
+  journeys: EnhancedJourney[],
+  dispatch: (action: fetchAction) => void
+): Promise<EnhancedJourney[]> => {
+  const { route, origin, direction, date } = input;
+
+  dispatch({ type: "FETCH_STARTED" });
+
+  return Promise.all(
+    journeys
+      .filter(journey => journey.realtime.prediction !== null)
+      .map(async journey => {
+        // Sometimes using the route.id from the journey is desired over the route.id
+        // from input (e.g. so we can get trips for "Green-D" instead of "Green")
+        const adjustedRoute = journey.route.id;
+
+        const tripId = journey.trip.id;
+
+        const tripInfo = await await fetch(
```

Comment:
It seems like `TableRow` is already requesting this data. Does that mean that we are now requesting it twice? If I'm getting that right that doesn't seem like something we'd want.

I'm trying to follow the changes in the diff syntax… is this data required for something at this level that can't be done inside of `TableRow`?

Reply:
Before this PR: for upcoming departures, the trip details call (which contains the crowding information) is made when expanding the row.
Since we need the crowding information before even expanding in order to show it in the row header, the big change in this PR is to make the above calls at the time when we retrieve the list of upcoming departures.
If you look at the network calls in the browser's devtools' network tab, expanding the rows will not make extra calls.
The trip information is already available and it's simply displayed.
This behavior only applies to Upcoming Departures, the behavior hasn't changed for the service selector (bottom part in the same modal).
Let me know if this makes sense or there's something still unclear.

Reply:
Thanks for the explanation. I think I was confusing the other `TableRow` component for the one that is local in this file, but makes more sense now.